### PR TITLE
[AD-318] Introduce base-noprelude and use Universum by default

### PR DESCRIPTION
**Pull request by goodsoft**
**YT issue:** https://issues.serokell.io/issue/AD-318

**Checklist:**

- [x] Updated docs if necessary
  - [x] [README](README.md)
  - [x] [TUI usage guide](docs/usage-tui.md)
  - [x] [Configuration documentation](docs/configuration.md)
- [ ] Adressed HLint warnings and hints
- [ ] Rebased branch on current master and squashed all "Address PR comment" commits
- [x] Tested my changes if they modify code

**Description:**
Knit still uses default Prelude as-per @int-index's requests.

### CODE REVIEW

File: ui/vty-lib/src/Ariadne/UI/Vty/Widget/Form/Edit.hs [hunk 1]
```diff
@@ -278,3 +275,10 @@ smartBreakLine :: TextZipper Text -> TextZipper Text
 smartBreakLine tz =
   let indentation = T.takeWhile isSpace (TZ.currentLine tz)
   in TZ.insertMany indentation (TZ.breakLine tz)
+
+until :: (a -> Bool) -> (a -> a) -> a -> a
```
Comment (Pastafarianist) [inline ui/vty-lib/src/Ariadne/UI/Vty/Widget/Form/Edit.hs]: That's necessary evil, I guess.

File: ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Accounts.hs [hunk 2]
```diff
@@ -6,13 +6,11 @@ module Ariadne.Wallet.Cardano.Kernel.Accounts (
...
     ) where
 
-import qualified Prelude
-import Universum
-
 import Control.Monad.Error.Class (liftEither, throwError)
 import qualified Data.Text.Buildable
 import Formatting (bprint, build, formatToString, (%))
 import qualified Formatting as F
+import qualified Text.Show
```
[the remark this answers is not in the document]
Comment (gromakovsky) [reply ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Accounts.hs]: It's needed to define a custom `Show` instance. Without it `show` is `Universum.show`, not `show` from the type class.

File: hpack/definitions.yaml [hunk 3]
```diff
@@ -54,7 +53,8 @@ _definitions:
         - ViewPatterns
 
     - &dependencies
-        - base >= 4.7 && < 5
+        - base-noprelude >= 4.7 && < 5
+        - loot-prelude
         - universum
```
Comment (gromakovsky) [inline hpack/definitions.yaml]: Do we need both `loot-prelude` and `universum`?

File: knit/package.yaml [hunk 4]
```diff
@@ -10,6 +10,8 @@ dependencies:
 
 library:
   <<: *lib-common
+  default-extensions:
+    - NoImplicitPrelude
```
Comment (gromakovsky) [inline knit/package.yaml]: How does it work for `knit`?

**Pull request merged**